### PR TITLE
Splitting actions into registry and proxy

### PR DESCRIPTION
**Pull request by arschles**
This patch logically splits the registry and proxy actions apart, and forces you to configure Athens as a proxy or a registry, but not both.

Fixes https://github.com/gomods/athens/issues/46

### CODE REVIEW

File: actions/app.go [hunk 1]
```diff
@@ -88,25 +79,22 @@ func App() *buffalo.App {
...
-		app.GET("/{base_url:.+}/{module}/@v/{version}.zip", versionZipHandler(storage))
-		app.POST("/admin/upload/{base_url:[a-zA-Z./]+}/{module}/{version}", uploadHandler(storage))
-		app.POST("/admin/fetch/{base_url:[a-zA-Z./]+}/{owner}/{repo}/{ref}/{version}", fetchHandler(storage))
-
-		auth := app.Group("/auth")
-		auth.GET("/{provider}", buffalo.WrapHandlerFunc(gothic.BeginAuthHandler))
-		auth.GET("/{provider}/callback", authCallback(mgoStore))
-		//	app.GET("/{base_url:.+}/{module}", homeHandler)
-		// serve files from the public directory:
+		if MODE == "proxy" {
```
Comment (michalpristas) [inline actions/app.go]: we have default set to proxy proxy should be also when invalid option is passed.
i would do explicit check for registry if(registry) else ( is proxy does not matter on value)

File: actions/app_registry.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+func addRegistryRoutes(app *buffalo.App) error {
+	cdnGetter := newCDNGetter()
+	mgoStore := mongo.NewMongoUserStore("127.0.0.1:27017")
+	if err := mgoStore.Connect(); err != nil {
+		return err
+	}
+	// serve go-get requests
+	app.Use(GoGet(cdnGetter))
+	auth := app.Group("/auth")
+	auth.GET("/{provider}", buffalo.WrapHandlerFunc(gothic.BeginAuthHandler))
```
Comment (michalpristas) [inline actions/app_registry.go]: shouldnt this be in opposite order?
auth + store (*Handler) - for registry
goGet - for proxy (without auth) 
or is auth needed for getting sources? please ping me to explain. my brain is in friday mode

Comment (arschles) [reply actions/app_registry.go]: not sure what you mean - the middleware intercepts all requests that came from `vgo get` - those can't require authentication

File: actions/app.go (hunk 1 above)
Comment (arschles) [reply actions/app.go]: done - I added a check for unsupported mode

**Comment by arschles** [comment]
I am going to merge this now and make further modifications in follow-ups

**Pull request merged**